tests/utils.py

```diff
@@ -27,7 +27,7 @@
 
 
 class HttpServerThread(Thread):
-    def __init__(self, handler: type[BaseRequestHandler]) -> None:
+    def __init__(self, handler: type[BaseRequestHandler], port: int = 0) -> None:
```

Comment:
I prefer having this one as a keyword argument because if you don't know that the second argument is the port, you might assume that it's a timeout (since it's a thread) or something else.